src/charts/raster-chart.js

```diff
@@ -513,6 +544,106 @@ export default function rasterChart(parent, useMap, chartGroup, _mapboxgl) {
         _x.domain([renderBounds[0][0], renderBounds[1][0]])
         _y.domain([renderBounds[2][1], renderBounds[0][1]])
       }
+    } else if (_chart.useTwoYAxes()) {
+      const layers = _chart.getLayers()
+      const xRanges = []
+      const yRanges = []
+      const y2Ranges = []
+
+      for (const layer of layers) {
+        let xDim = layer.xDim(),
+          yDim = layer.yDim()
+        if (xDim) {
+          let range = xDim.getFilter()
+          if (range !== null) {
+            // value coming back is an unnecessarily nested array - [[0, 1]]
+            if (range[0] && Array.isArray(range[0])) {
+              xRanges.push(range.flatMap(r => r))
+            } else {
+              xRanges.push(range)
+            }
+          }
+        }
+        if (yDim) {
+          let range = yDim.getFilter()
+          if (range !== null) {
+            // value coming back is an unnecessarily nested array - [[0, 1]]
+            if (range[0] && Array.isArray(range[0])) {
+              range = range.flatMap(r => r)
+            }
+            layer.layerType() === "crossSectionTerrain"
+              ? y2Ranges.push(range)
+              : yRanges.push(range)
+          }
+        }
+      }
+
+      if (xRanges.length) {
```

Comment:
If this is a direct copy/pasta from `_updateXAndYScales` it would probably be better to move this logic to a function so that changes don't need to be made in two places down the line.

Reply:
I actually am debating if the second part (`else` block) is used at all, and if it needs to stay.  Looking at the code, it really shouldn't even run - there are several problems:

- `getLayers()` is not declared as a stand-alone function anywhere in this file - it should be `_chart.getLayers()`
- `for (layer in layers)` isn't valid js - `layer` needs to be declared
- `viewBoxDim` isn't declared anywhere but is assigned a value

Maybe some of that stuff would pass in non-strict mode or whatever, but I don't see how the `getLayers()` call would work.  I was debating pulling it out entirely, but you know me and my extreme risk-adverse nature....